src/footer/footer.tsx

```diff
@@ -115,7 +116,11 @@ export const Footer = <T,>({
         }
 
         if (component) {
-            return <TopSection type="grid">{component}</TopSection>;
+            return (
+                <TopSection type="grid" stretch={layout === "stretch"}>
```

Comment:
can we have an assigned const like `isStretch` instead of doing the condition check each time.